File: src/main/java/com/platformbuilders/controleescolar/studentaccess/api/dto/StudentAccessDTO.java
package com.platformbuilders.controleescolar.studentaccess.api.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class StudentAccessDTO {
    private Long studentId;
    private LocalDateTime entryTime;
    private LocalDateTime exitTime;
    private String status;
}

File: bff/src/student-access/dto/student-access.dto.ts
import { Field, Int, ObjectType } from '@nestjs/graphql';

@ObjectType()
export class StudentAccessDTO {
  @Field(() => Int)
  id?: number;

  @Field(() => Int)
  studentId: number;

  @Field({ nullable: true })
  entryTime?: string;

  @Field({ nullable: true })
  exitTime?: string;

  @Field({ nullable: true })
  status?: string;
}